work around a compile error on openSUSE 11.0 and later.
See http://gcc.gnu.org/bugzilla/show_bug.cgi?id=11108

This is the workaround given there.

poeml, Sat Jul 12 13:44:48 CEST 2008

diff -uNrp metalink-0.3.3.orig/src/String/toUpper.cc metalink-0.3.3/src/String/toUpper.cc
--- metalink-0.3.3.orig/src/String/toUpper.cc	2006-11-29 20:14:45.000000000 +0100
+++ metalink-0.3.3/src/String/toUpper.cc	2008-07-12 13:44:08.472249000 +0200
@@ -1,5 +1,5 @@
 #include "String.ih"
 void String::toUpper()
 {
-	transform(this->begin(), this->end(), this->begin(), toupper);	
+	transform(this->begin(), this->end(), this->begin(), static_cast<int (*)(int)>(std::toupper));
 }
